packages/medusa/src/utils/authenticate-middleware.ts

```diff
@@ -104,6 +97,96 @@ export const authenticate = (
   }
 }
 
+const getApiKeyInfo = async (req: MedusaRequest): Promise<ApiKeyDTO | null> => {
+  const authHeader = req.headers.authorization
+  if (!authHeader) {
+    return null
+  }
+
+  const [tokenType, token] = authHeader.split(" ")
+  if (tokenType.toLowerCase() !== "bearer" || !token) {
```

Comment:
thought(non-blocking): Regarding your comment on Basic vs Bearer vs Something Third, I wanted to link a [discussion](https://github.com/medusajs/medusa/discussions/4037) opened in the community some time ago. 

This discussion actually prompted us to add a specific header `x-medusa-access-token` for API key authentication, instead of the Bearer which commonly carries a JWT.

However, I like the idea of moving back to Bearer or Basic for API key authentication, because those are native authentication headers and are therefore by default handled appropriately by clients and proxies, e.g. avoiding any type of caching. 

Would be curios to hear what you think?

Reply:
@olivermrbl ok went through that discussion, I think some points are valid.

However, one thing that people miss about Bearer tokens is that they are not defined anywhere on what exactly they should carry. The do commonly carry JWT tokens as Bearer was added for Oauth 2.0, but they can literally carry anything that says "Give access to the bearer of this token". 

Now, I think it's still good to align with what people expect, rather than what is technically correct. I checked Stripe and they suggest Basic auth with the token, which I think is fine and we can do. This will just need another middleware to extract the basic credentials before calling the auth middleware, but it's not a big deal.

When it comes to session vs JWT authentication, it all depends on which tradeoffs you want to take. I think a pure cookie-based session auth is a bit outdated as it limits how this API can be called from eg. mobile apps (as that discussion points out). However, if you do want to allow revoking JWT tokens, for example, or you need a fine-grained authorization management, you'll end up with some session state regardless, so JWT token authentication gets a different meaning. That, however, is something we can discuss at a later point.

For now, I can just follow Stripe's approach and just use basic auth?